### ISSUE TITLE
Adjust YAW direction to destination travelnet orientation

### ISSUE THREAD

**Issue by bark10** [Sokomine/travelnet#18]
It would be nice if the players view direction was corrected to the destination travelnet's exit. Currently the player will be facing the same direction at the destination-travelnet as on the origin-travelnet, which often results in facing one of the three travelnet walls instead of the exit.

Perhaps you've already looked into this. It would surely be an improvement.

**Comment by HybridDog** [comment]
l agree, this can easily be done.

**Comment by bark10** [comment]
My apologies, I had not read the changelog and code thoroughly enough. It turns out this feature is already present in the mod. And it does indeed work, just not consistently.

Let me rephrase this as a bug report. When using a travelnet, the yaw is usually set as intended, but not always. It appears to be independent of what station I'm travelling to, as a given station may have a wrong yaw on my first travel there, but correct on the second or third travel or vice versa.

I have replicated the bug on my server, a test single player game and on other public servers that are using travelnets.

I have not been able to replicate the bug in a small network of 3 travelnets. This also appears to be the case if the travelnets are close to each other.

I noticed in the [minetest lua api](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt) that set_look_pitch and set_look_yaw has been deprecated, so I tried changing this in the init.lua file.

>#set_look_pitch(radians): sets look pitch - Deprecated. Use set_look_vertical.
>#set_look_yaw(radians): sets look yaw - Deprecated. Use set_look_horizontal.

Changing this did not make a difference.

**Comment by HybridDog** [comment]
https://github.com/Sokomine/travelnet/blob/master/init.lua#L559
bark10, l assume it happens when you teleport to a station in an unloaded chunk. You can use vmanip to load the node there:
```
if node2.name == "ignore" then
	minetest.get_voxel_manip():read_from_map(pos, pos)
	node2 = minetest.get_node(pos)
end
```

And indeed, look_yaw and look_pitch are deprecated, they have a strange behaviour.

**Comment by Sokomine** [comment]
The deprecated functions have been updated.
The target travelnet's orientation is indeed not stored but read from the map. Thus, players will arrive with wrong rotation at nodes that are currently unloaded. Force-loading that node would be possible, but is it really desirable? Voxelmanip tends to have some overhead. It's small, it's just one node, but still - some server owners are very concerned about performance. I want travelnet's footprint on CPU ressources to be low.

**Comment by HybridDog** [comment]
> Voxelmanip tends to have some overhead. It's small, it's just one node, but still - some server owners are very concerned about performance. I want travelnet's footprint on CPU ressources to be low.

Voxelmanip works very fast. You can set a mapblock to stone more than 30 times a second using worldedit without causing noticeable lag.
Also, when the player teleports there, the blocks become loaded anyway.

**Comment by Sokomine** [comment]
Server owners are really very reluctant as far as anything that might cause lag goes.

The stations now try harder to rotate the player correctly. If the target node isn't loaded, it'll wait a moment and try again.

Also normal doors ought to open correctly again.

**Comment by Sokomine** [comment]
Fixed in latest release as far as can be.

**Comment by HybridDog** [comment]
Since minetest 5.0.0, there's the minetest.load_area function.
